### PR TITLE
[improve](fmt) use format_to and FMT_COMPILE to speed up write data

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: be/src/runtime/fold_constant_executor.cpp [hunk 1]
```diff
@@ -170,22 +170,22 @@ Status FoldConstantExecutor::_get_result(void* src, size_t size, const TypeDescr
     }
     case TYPE_TINYINT: {
         int8_t val = *reinterpret_cast<const int8_t*>(src);
-        result = fmt::format_int(val).str();
+        fmt::format_to(result.data(), FMT_COMPILE("{}"), val);
```
Comment (HappenLee) [inline be/src/runtime/fold_constant_executor.cpp:173]: result had already alloc mem to write？

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zclllyybb — APPROVED** [review]
LGTM

**Pull request merged**